packages/store/src/entity-base.ts

```diff
@@ -28,41 +29,68 @@ export enum EntityMutation {
   None
 }
 
-export abstract class EntityBase<T, S extends EntityState<T>> {
+export abstract class EntityBase<T, S extends EntityStateModel<T>> {
+  // @todo Don't like having to pass in an injector to be able to get at the
+  // state stream, how could we do this differently?
+  private _stateStream: StateStream;
```

Comment:
What if we do what we did for `@Select` decorators for this? Make a static instance u can point to - https://github.com/ngxs/store/blob/master/packages/store/src/select.ts#L13